src/price-spread-ratio/price-spread-ratio.spec.js

```diff
@@ -0,0 +1,244 @@
+    expect(axios.get).toHaveBeenCalledTimes(1);
+    expect(response.data.result.price).toEqual(42);
+  });
+});
+
+/* agent tests */
+describe('price spread ratio monitoring', () => {
+  let initialize;
+  let handleBlock;
+  let data;
+  let blockEvent;
+
+  // for the tests, we will keep the mocked FTX API price constant and change the mocked price
+  // returned by the perpetual finance contract
+  const mockFtxPrice = 100;
+  const upperLimitPercent = 1;
+  const lowerLimitPercent = -1;
+  const mockContractName = 'FAKE1';
+  const mockContractAddress = '0xFAKEADDRESS1';
+
+  beforeEach(async () => {
+    axios.get.mockClear();
+
+    // reset the price returned by the mocked FTX API
+    mockFtxResponse.data.result.price = mockFtxPrice;
+
+    data = {};
+
+    initialize = provideInitialize(data);
+    await initialize();
+
+    // initialize the data structure
+    data.provider = {};
+    data.contractNames = [
+      mockContractName,
+    ];
+    data.contractsData = [{
+      name: mockContractName,
+      address: mockContractAddress,
+      timeThresholdSeconds: BigInt(100),
+    }];
+
+    // set up the upper and lower limits to make the price spread ratio acceptable
+    data.contractsData[0].upperLimitPercent = new BigNumber(upperLimitPercent);
+    data.contractsData[0].lowerLimitPercent = new BigNumber(lowerLimitPercent);
+
+    // create the handler
+    handleBlock = provideHandleBlock(data);
+
+    // create a block event that will initialize the timestamp on the contract data object
+    blockEvent = {
+      block: {
+        timestamp: 1,
+      },
+    };
+
+  });
+
+  // reset function call count after each test
+  afterEach(() => {
+    axios.get.mockClear();
+  });
+
+  describe('handleBlock', () => {
+    it('returns empty findings if the price spread ratio is within the limits', async () => {
+      // set up prices that will not trigger an alert
+      const mockPerpPrice = new BigNumber(101);
+      const mockPerpPriceWei = mockPerpPrice.times(WEI_PER_ETHER);
+
+      // set up the mock contract methods to return the price information
+      data.contractsData[0].contract = {
+        getIndexPrice: jest.fn().mockResolvedValue(mockPerpPriceWei),
+      };
+
+      // run the handler
+      let findings = await handleBlock(blockEvent);
+
+      // check assertions
+      expect(findings).toStrictEqual([]);
+
+      // intentionally set the next timestamp to be beyond the time threshold
+      blockEvent.block.timestamp += (1 + Number(data.contractsData[0].timeThresholdSeconds));
+
+      // run the handler again
+      findings = await handleBlock(blockEvent);
+
+      // check assertions
+      // axios.get will not be called for the first run of the handler
+      expect(axios.get).toHaveBeenCalledTimes(1);
+      expect(findings).toStrictEqual([]);
+    });
+
+    it('returns empty findings if the price spread ratio it beyond the threshold but not enough time has elapsed', async () => {
```

Comment:
typo: "...price spread ratio _is_ beyond the threshold..."

Reply:
Good catch!  Fixed.